### DISCUSSION TITLE
Table instead of cards for task overview

### DISCUSSION THREAD

**Discussion by JJ-8** [TFNS/CTFNote#102] [Ideas]
I get often the feedback that the task overview of a CTF is a bit messy. There are lots of cards of different sizes and colors all over the screen which makes selecting your task a _difficult_ task.

People have suggested to me to change the cards to a table and display all the information in there. So something like this:

| Title | Description | Category | On it | Flag submit | Config buttons |
| --- | --- | --- | --- | --- | --- |

> The description column would be truncated after X characters and the user can click on the description to see a dialog with the complete description

This would make navigation easier for them. It is similar to the old menu in CTFNote, but with some more information and in a real table:
![Old CTFNote menu](https://user-images.githubusercontent.com/34778827/138764939-223d9078-1d58-4b2e-a461-29a9a41728d3.png)

Some additional functionalities that would make navigation easier:
- Display the challenges that the current player is working on first. So this would make the "On it" feature also a "pin to top" feature. This could prevent scrolling on the overview page and motivate people to use the "On it" functionality.
- Display the solved challenges at the end. So as soon as a flag has been submitted, the challenge would automatically go to the bottom of the table so the user doesn't have to "unpin" the challenge (and therefore losing the information of who had worked on it).
- For the general sort, sort by category just like the current behavior of CTFNote.

The current overview doesn't have to be replaced, but maybe a new display mode "table" can be added next to the "classic", "dense" and "ultradense" display modes.

**Comment by B-i-t-K** [comment]
I think we can do a little cleaning on the TaskList component and the different display modes.
One component for each display mode, and add a table display.

I like the idea of pining the tasks a user is working on also

**Comment by XeR** [comment]
I like the idea of having a table to display tasks.

> Display the challenges that the current player is working on first.
> Display the solved challenges at the end.

I agree. I usually hide solved tasks because they get in the way.


>  maybe a new display mode "table" can be added next to the [...] display modes.

I agree. This looks like the best way to do it.

I'm not sure about the column order… we'll see that bit later.

**Comment by JJ-8** [comment]
This has been implemented for quite a while now, so I will close this discussion.
